File: 프로그래머스/1/142086. 가장 가까운 같은 글자/가장 가까운 같은 글자.py
def solution(s):
    # 딕셔너리를 만들건데
    answer = []
    key = {}
    for i, ch in enumerate(s):
        if ch not in key:
            key[ch] = i
            answer.append(-1)
        else:
            answer.append(i - key[ch])
            key[ch] = i
    return answer

File: 프로그래머스/1/142086. 가장 가까운 같은 글자/가장 가까운 같은 글자.js
function solution(s) {
    let answer = [];
    let dict = {};
    for (let i = 0; i < s.length; i++) {
        s[i] in dict ? answer.push(i - dict[s[i]]) : answer.push(-1);
        dict[s[i]] = i
    }
    return answer;
}